src/utils/general.utils.ts

```diff
@@ -231,8 +231,57 @@ export function formatAmountWithDecimals({ amount, decimals }: { amount: number;
     return formattedAmount
 }
 
-export function formatAmount(amount: number) {
-    return amount.toFixed(2)
+/**
+ * formats a number by removing unnecessary trailing zeros after decimal point
+ * if all decimal places are zeros, returns the whole number
+ * preserves small decimal numbers
+ * @param amount - number or string to format
+ * @returns formatted string representation of the number
+ */
+export const formatAmount = (amount: string | number): string => {
+    // handle undefined, null, or empty string
```

Comment:
Issue, blocking: It'd be nice to test this function under various examples. Can we add test cases for it? Should be simple, we already have jest setup